src/core/coap/coap_secure.hpp

```diff
@@ -53,13 +53,10 @@ class CoapSecure : public CoapBase
 {
 public:
     /**
-     * Pointer is called once DTLS connection is established.
-     *
-     * @param[in]  aConnected  TRUE if a connection was established, FALSE otherwise.
-     * @param[in]  aContext    A pointer to arbitrary context information.
+     * Function pointer which is called reporting a connection event (when connection established or disconnected)
      *
      */
-    typedef void (*ConnectedCallback)(bool aConnected, void *aContext);
+    typedef otHandleCoapSecureClientConnect ConnectedCallback;
```

Comment:
The type `otHandleCoapSecureClientConnect` is for a connection event which is not exclusively for 'connected' event. Could you rename the type alias to something like `ConnectEventCallback`?

@abtink What do you think?

Reply:
Updated the alias and also the `otCoapSecureSetClientConnectedCallback` api